src/xmagics/execution.cpp

```diff
@@ -85,21 +93,24 @@ namespace xcpp
         // std::vector<std::string> results((std::istream_iterator<std::string>(iss)),
         //                          std::istream_iterator<std::string>());
 
-        auto options = get_options();
-        auto result = options.parse(line);
+        auto argpars = get_options();
+        argpars.parse(line);
 
-        std::size_t number = (result.count("n")) ? result["n"].as<std::size_t>() : 0ul;
-        std::size_t repeat = result["r"].as<std::size_t>();
-        std::size_t precision = result["p"].as<std::size_t>();
+        // TODO find a way to use std::size_t
+        int number = argpars.get<int>("-n");
+        int repeat = argpars.get<int>("-r");
+        int precision = argpars.get<int>("-p");
 
         std::string code;
-        if (result.count("positional"))
-        {
-            auto& v = result["positional"].as<std::vector<std::string>>();
+        try {
+            const auto& v = argpars.get<std::vector<std::string>>("expression");
             for (const auto& s : v)
             {
                 code += " " + s;
             }
+        } catch (std::logic_error& e) {
+            if (trim(cell).empty())
```

Comment:
Good practice: always ue curly braces for if statements, even if the body of the if is a single line